src/store/actions/contact.actions.ts

```diff
@@ -34,7 +34,7 @@ import store from '../index';
 export const getContacts = (): ThunkAction<Promise<void>, RootState, unknown, Action> => {
   return async (dispatch: ThunkDispatch<RootState, unknown, Action>) => {
     dispatch({type: CONTACTS_LOADING});
-    getUserContact().then((userContact: IContact) => {
+    await getUserContact().then((userContact: IContact) => {
```

Comment:
why are we now awaiting this? should the concept not be that you dispatch an action and you should not care when it completes? at least that is my understanding from redux actions

Reply:
Yeah you are correct, will have to think about this one, as it also means the setupWallet state would then finish before it is actually done and thus the machine would progress to the end state. I really want the state machine to be responsible for the progression to the next state and navigation and only do that once it is done.